tfhe/src/shortint/engine/server_side/scalar_mul.rs

```diff
@@ -47,6 +47,9 @@ impl ShortintEngine {
         ctxt: &mut Ciphertext,
         scalar: u8,
     ) -> EngineResult<Ciphertext> {
+        if !server_key.is_scalar_mul_possible(ctxt, scalar) {
+            self.message_extract_assign(server_key, ctxt)?;
+        }
         let mut ct_result = ctxt.clone();
         self.smart_scalar_mul_assign(server_key, &mut ct_result, scalar)?;
```

Comment:
unchecked call ? not sure if the function does other funky stuff

Reply:
Even with a clean ct, the multiplication might not pass, is that right?

Anyway, the cost of calling smart instead of unchecked should be negligible

Reply:
fair enough